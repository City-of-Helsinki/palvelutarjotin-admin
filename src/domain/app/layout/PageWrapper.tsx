import classNames from 'classnames';
import * as React from 'react';
import { Helmet } from 'react-helmet';
import { useTranslation } from 'react-i18next';
import { useLocation } from 'react-router-dom';

import styles from './pageWrapper.module.scss';
import useLocale from '../../../hooks/useLocale';

interface Props {
  className?: string;
  title?: string;
  children?: React.ReactNode;
}

const PageWrapper: React.FC<Props> = ({
  children,
  className,
  title = 'appName',
}) => {
  const { t } = useTranslation();
  const locale = useLocale();
  const { pathname } = useLocation();

  const translatedTitle =
    title !== 'appName' ? `${t(title)} - ${t('appName')}` : t('appName');

  let path = pathname.replace(`/${locale}`, '');
  path = path.startsWith('/') ? path.slice(1) : path;

  return (
    <div className={classNames(styles.pageWrapper, className)}>
      <Helmet>
        <html lang={locale} />
        <title>{translatedTitle}</title>
        <link rel="alternate" hrefLang="fi" href={'/fi/' + path} />
        <link rel="alternate" hrefLang="sv" href={'/sv/' + path} />
        <link rel="alternate" hrefLang="en" href={'/en/' + path} />
      </Helmet>
      {children}
    </div>
  );
};

export default PageWrapper;
