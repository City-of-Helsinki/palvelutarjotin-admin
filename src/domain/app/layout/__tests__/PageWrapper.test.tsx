import { shallow } from 'enzyme';
import pretty from 'pretty';
import * as React from 'react';
import { MemoryRouter } from 'react-router';

import PageWrapper from '../PageWrapper';

it('PageWrapper matches snapshot', () => {
  const pageWrapper = shallow(
    <MemoryRouter initialEntries={['/']}>
      <PageWrapper />
    </MemoryRouter>
  );
  expect(pretty(pageWrapper.html())).toMatchSnapshot();
});
