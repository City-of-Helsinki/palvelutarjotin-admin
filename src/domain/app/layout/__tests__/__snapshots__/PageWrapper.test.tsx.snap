// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageWrapper matches snapshot 1`] = `
<div
  class="pageWrapper"
/>
`;
